pkg/clusteragent/admission/mutate/common/common.go

```diff
@@ -11,29 +11,33 @@ package common
 import (
 	"encoding/json"
 	"fmt"
+	"strconv"
 
 	"github.com/wI2L/jsondiff"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/client-go/dynamic"
 
 	admCommon "github.com/DataDog/datadog-agent/pkg/clusteragent/admission/common"
+	"github.com/DataDog/datadog-agent/pkg/clusteragent/admission/metrics"
 	"github.com/DataDog/datadog-agent/pkg/config"
 	"github.com/DataDog/datadog-agent/pkg/util/log"
 )
 
 // MutationFunc is a function that mutates a pod
-type MutationFunc func(*corev1.Pod, string, dynamic.Interface) error
+type MutationFunc func(*corev1.Pod, string, dynamic.Interface) (bool, error)
 
 // Mutate handles mutating pods and encoding and decoding admission
 // requests and responses for the public mutate functions
-func Mutate(rawPod []byte, ns string, m MutationFunc, dc dynamic.Interface) ([]byte, error) {
+func Mutate(rawPod []byte, ns string, mutationType string, m MutationFunc, dc dynamic.Interface) ([]byte, error) {
 	var pod corev1.Pod
 	if err := json.Unmarshal(rawPod, &pod); err != nil {
 		return nil, fmt.Errorf("failed to decode raw object: %v", err)
 	}
 
-	if err := m(&pod, ns, dc); err != nil {
-		return nil, err
+	if injected, err := m(&pod, ns, dc); err != nil {
+		metrics.MutationErrors.Inc(mutationType, err.Error())
+	} else {
+		metrics.MutationAttempts.Inc(mutationType, strconv.FormatBool(injected))
```

Comment:
I was thinking that we always increment attempts even if the attempts is an error. 

imo we can reduce the number of metrics and juste use `datadog.cluster_agent.admission_webhooks.mutation_attempts` and removed `datadog.cluster_agent.admission_webhooks.mutation_errors`

what we can do is to add the `status` tag to the metric `status:success` or `status:error` and maybe add `error:<error-type>` only when the `status:error`

Reply:
I agree.